### PR TITLE
{Compute} Adopt track2 SDK, azure-mgmt-batch==15.0.0

### PR DESCRIPTION
**Pull request by BigCat20196**
**Description**

Here are main changes for Servicebus migration to track2:

- LRO operations are renamed to begin_xxx_xxx, for example in track1, the name is delete; while in track2, the name is `begin_delete.
- For generic update of LRO, set setter_name to begin_create_or_update.
- In track2, SDK parameters are not flattened, so we need pass object parameter instead of flattened parameters to SDK.
- Fix some tests due to changes in track2.
- Run live tests and re-record tests.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/batch/_params.py [hunk 1]
```diff
@@ -95,10 +101,12 @@ def load_arguments(self, _):
...
-    with self.argument_context('batch application package create') as c:
-        c.argument('package_file', type=file_type, help='The path of the application package in zip format', completer=FilesCompleter())
-        c.argument('application_name', options_list=('--application-name',), help="The name of the application.")
-        c.argument('version_name', options_list=('--version-name',), help="The version name of the application.")
+    for command in ['create', 'activate']:
+        with self.argument_context('batch application package {}'.format(command)) as c:
+            c.argument('package_file', type=file_type, help='The path of the application package in zip format', completer=FilesCompleter())
+            c.argument('application_name', options_list=('--application-name',), help="The name of the application.")
+            c.argument('version_name', options_list=('--version-name',), help="The version name of the application.")
+            c.argument('f_ormat', options_list=('--format',), help="The format of the application package binary file.")
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/batch/_params.py:109]:
```suggestion
            c.argument('format', options_list=('--format',), help="The format of the application package binary file.")
```

Comment (BigCat20196) [reply src/azure-cli/azure/cli/command_modules/batch/_params.py:109]: Format is a keyword. If we use format as the parameter name, linter will report an error.

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/batch/_params.py:109]: You are right. Options_list is the display name. No problem.

**Review by qwordy — APPROVED** [review]

**Comment by qwordy** [comment]
One failed.
https://clitestresultstac.blob.core.windows.net/20210528-025442-145921/batch.report.parallel.html

**Comment by BigCat20196** [comment]
Yeah, there is a historical error in this module, and it is not caused by Azure-cli.  

I opened an issue(#18208 ) and asked for help from the service team.

**Comment by qwordy** [comment]
@jsntcy, @kairu-ms, @msyyc could you help review this PR?

**Review by kairu-ms — APPROVED** [review]

**Review by jsntcy — APPROVED** [review]

**Pull request merged**